flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/internals/fanout/FanOutProperties.java

```diff
@@ -0,0 +1,127 @@
+package org.apache.flink.streaming.connectors.kinesis.internals.fanout;
+
+import org.apache.flink.streaming.connectors.kinesis.config.ConsumerConfigConstants;
+import org.apache.flink.streaming.connectors.kinesis.config.ConsumerConfigConstants.EFORegistrationType;
+import org.apache.flink.streaming.connectors.kinesis.config.ConsumerConfigConstants.RecordPublisherType;
+import org.apache.flink.streaming.connectors.kinesis.util.KinesisConfigUtil;
+import org.apache.flink.util.Preconditions;
+
+import javax.annotation.Nullable;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+
+
+/**
+ * This is a configuration class for enhanced fan-out components.
+ */
+public class FanOutProperties implements Serializable {
+	private static final long serialVersionUID = 3204635913413261619L;
+
+	private EFORegistrationType efoRegistrationType;
+	@Nullable
+	private String consumerName;
+	@Nullable
+	private List<String> streamConsumerArns;
+
+	private int subscribeToShardMaxRetries;
+
+	private long subscribeToShardMaxBackoffMillis;
+
+	private long subscribeToShardBaseBackoffMillis;
+
+	private double subscribeToShardExpConstant;
+
+	public FanOutProperties(Properties properties, List<String> streams) {
+		//validate the properties
```

Comment:
@xiaolong-sn some of your comments might violate the "golden rule", ["Golden rule: Comment as much as necessary to support code understanding, but don’t add redundant information."](https://flink.apache.org/contributing/code-style-and-quality-common.html#comments-and-code-readability)

Reply:
Yeah, I see your point. The comments are redundant.